### PR TITLE
Cy 789 enable ssl communication with postgre sql in an external db scenario

### PR DESCRIPTION
**Pull request by barucoh**

### CODE REVIEW

File: cfy_manager/components/postgresql_server/postgresql_server.py [hunk 1]
```diff
@@ -110,6 +122,17 @@ def _write_new_pgconfig_file(self, lines):
                     line = line.replace('#listen_addresses = \'localhost\'',
                                         'listen_addresses = \'{0}\''
                                         .format(config[MANAGER][PRIVATE_IP]))
+                if config[POSTGRESQL_SERVER][SSL_ENABLED]:
+                    if line.startswith('#port = 5432'):
+                        line.replace('#port = 5432',
```
Comment (tehasdf) [inline cfy_manager/components/postgresql_server/postgresql_server.py]: you mean `line = line.replace(...)`

File: cfy_manager/components/postgresql_client/postgresql_client.py [hunk 2]
```diff
@@ -12,15 +12,24 @@
 #  * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 #  * See the License for the specific language governing permissions and
 #  * limitations under the License.
-
-from os.path import join
+import os
```
Comment (tyacbovi) [inline cfy_manager/components/postgresql_client/postgresql_client.py]: style - enter before this line

File: cfy_manager/components/postgresql_client/postgresql_client.py [hunk 3]
```diff
@@ -12,15 +12,24 @@
...
+import shutil
 
 from ...exceptions import ProcessExecutionError
 
-from ..components_constants import SOURCES
+from ..components_constants import (
```
Comment (tyacbovi) [inline cfy_manager/components/postgresql_client/postgresql_client.py]: style - move down two lines

File: cfy_manager/components/postgresql_server/postgresql_server.py [hunk 4]
```diff
@@ -47,9 +50,15 @@
...
+CA_CERT_PATH = os.path.join(os.path.dirname(PG_CONF_PATH), 'root.crt')
+SERVER_CERT_PATH = os.path.join(os.path.dirname(PG_CONF_PATH), 'server.crt')
+SERVER_KEY_PATH = os.path.join(os.path.dirname(PG_CONF_PATH), 'server.key')
+
 PG_HBA_LISTEN_ALL_REGEX_PATTERN = r'host\s+all\s+all\s+0\.0\.0\.0\/0\s+trust'
+PG_HBA_HOSTSSL_REGEX_PATTERN = \
+    r'hostssl\s+all\s+all\s+0\.0\.0\.0\/0\s+md5\s+clientcert=1'
 
-PG_PORT = 5432
+PG_PORT = 15432 if config[POSTGRESQL_SERVER][SSL_ENABLED] else 5432
```
Comment (tyacbovi) [inline cfy_manager/components/postgresql_server/postgresql_server.py]: Maybe put this in a func, so the port number will be set in one place and config will changed

File: cfy_manager/components/postgresql_server/postgresql_server.py (hunk 1 above)
Comment (barucoh) [reply cfy_manager/components/postgresql_server/postgresql_server.py]: Right, fixed

File: cfy_manager/components/postgresql_server/postgresql_server.py [hunk 5]
```diff
@@ -124,8 +147,30 @@ def _write_new_hba_file(self, lines, enable_remote_connections):
...
+            if config[POSTGRESQL_SERVER][SSL_ENABLED] and not \
+                    re.search(PG_HBA_HOSTSSL_REGEX_PATTERN, '\n'.join(lines)):
+                f.write('hostssl all all 0.0.0.0/0 md5 clientcert=1')
+
         return temp_hba_path
 
+    def _configure_ssl(self):
+        """
+        Copy SSL certificates to postgres data directory.
+        Configuration for postgresql.conf is done in _write_new_pgconfig_file
```
Comment (tyacbovi) [inline cfy_manager/components/postgresql_server/postgresql_server.py]: I would say "postgresql.conf and pg_hba.conf configurations are handled in the create configuration step"

File: cfy_manager/components/postgresql_server/postgresql_server.py [hunk 6]
```diff
@@ -124,8 +147,30 @@ def _write_new_hba_file(self, lines, enable_remote_connections):
...
+        Configuration for pg_hba.conf is done in _write_new_hba_file
+
+        The validation for the existence of the files is already done in
```
Comment (tyacbovi) [inline cfy_manager/components/postgresql_server/postgresql_server.py]: Not needed, it's clear

File: cfy_manager/components/validations.py [hunk 7]
```diff
@@ -355,10 +366,44 @@ def _check_internal_ca_cert():
...
 
 
+def _validate_postgres_ssl_certificates_provided():
+    error_msg = 'If Postgresql requires SSL communication {0} a ' \
+                'certificate and a key for Postgresql must be provided in ' \
+                'config.yaml->ssl_inputs->{1}'
+    if not (config[SSL_INPUTS][POSTGRESQL_SERVER_CERT_PATH] and
+            config[SSL_INPUTS][POSTGRESQL_SERVER_KEY_PATH]):
+        if config[POSTGRESQL_SERVER][SSL_ENABLED]:
+            raise ValidationError(error_msg.format(
```
Comment (tyacbovi) [inline cfy_manager/components/validations.py]: does not create a logical sentence

File: cfy_manager/components/validations.py [hunk 8]
```diff
@@ -355,10 +366,44 @@ def _check_internal_ca_cert():
...
+def _validate_external_postgres_ssl_enabled():
+    """
+    Basically, making sure that if the manager and database are not on the same
+    machine, SSL for DB communication must be enabled
+    :return:
+    """
+    if not config[POSTGRESQL_SERVER][SSL_ENABLED] and \
+            MANAGER_SERVICE not in config[SERVICES_TO_INSTALL] and \
+            DATABASE_SERVICE in config[SERVICES_TO_INSTALL] \
+            or \
```
Comment (tyacbovi) [inline cfy_manager/components/validations.py]: Can be combined to a single assert

File: cfy_manager/components/postgresql_client/postgresql_client.py (hunk 2 above)
Comment (barucoh) [reply cfy_manager/components/postgresql_client/postgresql_client.py]: fixed

File: cfy_manager/components/postgresql_client/postgresql_client.py (hunk 3 above)
Comment (barucoh) [reply cfy_manager/components/postgresql_client/postgresql_client.py]: fixed

File: cfy_manager/components/postgresql_server/postgresql_server.py (hunk 5 above)
Comment (barucoh) [reply cfy_manager/components/postgresql_server/postgresql_server.py]: fixed

File: cfy_manager/components/postgresql_server/postgresql_server.py (hunk 6 above)
Comment (barucoh) [reply cfy_manager/components/postgresql_server/postgresql_server.py]: Fair, fixed

File: cfy_manager/components/validations.py (hunk 8 above)
Comment (barucoh) [reply cfy_manager/components/validations.py]: Fixed

File: cfy_manager/components/postgresql_server/postgresql_server.py (hunk 4 above)
Comment (barucoh) [reply cfy_manager/components/postgresql_server/postgresql_server.py]: Fixed

**Pull request merged**